packages/yarn-plugin/src/resolver/BackstageResolver.ts

```diff
@@ -44,7 +44,10 @@ export class BackstageResolver implements Resolver {
    * the version in backstage.json changes.
    */
   bindDescriptor(descriptor: Descriptor): Descriptor {
-    if (descriptor.range !== 'backstage:^') {
+    if (
+      descriptor.range !== 'backstage:^' &&
+      !descriptor.range.startsWith('backstage:')
+    ) {
```

Comment:
Interesting that bindDescriptor can be called for entries in yarn.lock somehow! Curious to understand why that happens, but in any case - given this, I'd probably flip the whole thing to something like:

```typescript
if (descriptor.range === 'backstage:^') {
  return structUtils.makeDescriptor(
    descriptor,
    `${PROTOCOL}${getCurrentBackstageVersion()}`,
  );
}

return descriptor;
```

Reply:
I was even thinking we could do something like this instead: 

```ts
if (
      descriptor.range !== 'backstage:^' &&
      descriptor.range !== `${PROTOCOL}${getCurrentBackstageVersion()}`
    ) {
      throw new Error(
        `Unsupported version range "${
          descriptor.range
        }" for package ${structUtils.stringifyIdent(
          descriptor,
        )}. The backstage protocol only supports the range "backstage:^" or the current Backstage version "${PROTOCOL}${getCurrentBackstageVersion()}".`,
      );
    }

    return structUtils.makeDescriptor(
      descriptor,
      `${PROTOCOL}${getCurrentBackstageVersion()}`,
    );
```

Reply:
That way it throws if something is pointing to the wrong version?

Reply:
Hmm, was kinda thinking that being able to point to a different version for an individual is a feature though?

Reply:
Yeah, that was my thinking too

Reply:
Fixed and tests updated